Networking/Networking/Model/Product/Product.swift

```diff
@@ -266,17 +266,18 @@ public struct Product: Codable {
 
         let regularPrice = try container.decodeIfPresent(String.self, forKey: .regularPrice)
 
+        let onSale = try container.decode(Bool.self, forKey: .onSale)
+
         // Even though a plain install of WooCommerce Core provides string values,
         // some plugins alter the field value from String to Int or Decimal.
         var salePrice = ""
         if let parsedSalePriceString = container.failsafeDecodeIfPresent(stringForKey: .salePrice) {
-            salePrice = parsedSalePriceString
+            salePrice = (onSale && parsedSalePriceString.isEmpty) ? "0" : parsedSalePriceString
```

Comment:
I'm terrified of user errors and would also `trim` before checking `isEmpty`. Do you think that's necessary?

Reply:
I verified, and the backend trims it before saving it. 👍